packages/web-config-server/src/authSession/getUserFromAuthHeader.js

```diff
@@ -3,21 +3,24 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 import {
-  encryptPassword,
   getUserAndPassFromBasicAuth,
   getTokenClaimsFromBearerAuth,
+  verifyPassword,
 } from '@tupaia/auth';
 
 const getApiClientUserFromBasicAuth = async (models, authHeader) => {
   const { username, password: secretKey } = getUserAndPassFromBasicAuth(authHeader);
 
   // first attempt to authenticate as an api client, in case a secret key was used in the auth header
-  const secretKeyHash = encryptPassword(secretKey, process.env.API_CLIENT_SALT);
   const apiClient = await models.apiClient.findOne({
     username,
-    secret_key_hash: secretKeyHash,
   });
-  return apiClient?.getUser();
+  const verified = await verifyPassword(
+    secretKey,
+    process.env.API_CLIENT_SALT,
+    apiClient.secret_key_hash,
+  );
+  return verified ? apiClient?.getUser() : undefined;
```

Comment:
This could throw an error if no `apiClient` is found?
```suggestion
  if (!apiClient) {
    return undefined;
  }
  
  const verified = await verifyPassword(
    secretKey,
    process.env.API_CLIENT_SALT,
    apiClient.secret_key_hash,
  );
  return verified ? apiClient.getUser() : undefined;
```